rust/blockstore/src/arrow/block/value/data_record_value.rs

```diff
@@ -45,6 +67,116 @@ impl ArrowWriteableValue for &DataRecord<'_> {
     fn get_delta_builder() -> BlockStorage {
         BlockStorage::DataRecord(DataRecordStorage::new())
     }
+
+    fn get_arrow_builder(size_tracker: Self::SizeTracker) -> Self::ArrowBuilder {
+        ValueBuilderWrapper {
+            id_builder: StringBuilder::with_capacity(
+                size_tracker.get_num_items(),
+                size_tracker.get_id_size(),
+            ),
+            embedding_builder: FixedSizeListBuilder::with_capacity(
+                Float32Builder::with_capacity(
+                    size_tracker.get_num_items()
+                        * size_tracker.get_embedding_dimension().unwrap_or(0),
```

Comment:
should this really unwrap_or to 0?

Reply:
this is the same behavior as before, maybe it would be more correct to error if there are no items?